src/answers-headless.ts

```diff
@@ -66,7 +67,7 @@ export default class AnswersHeadless {
     this.stateManager.dispatchEvent('vertical/setOffset', offset);
   }
 
-  setFilter(filter: Filter | CombinedFilter | null): void {
+  setFilter(filter: Record<string, SelectableFilter[]> | null): void {
```

Comment:
Can we simplify this interface by having setFilter take in a single SelectableFilter? SelectableFilter would need to have it's own `filterSetId` or `filterGroupId`. If we want to allow an array, we could have a `setFilters` function

Reply:
from my understanding, `setFilter` purpose is to be able to change the entire `static` portion of `FilterState` previously so I kept the functionality as is. If user want to modify/add a new set of static filters (say another StaticFilter component), they would use `addFilters`. I could rename it to:
`setFilter(filter: Record<string, SelectableFilter[]> | null)` => `setFilters(...)`
`addFilters(filterSetId: string, filters: SelectableFilter[])` => `setFilter(...)`
so it's a bit more consistent. Does that help?

Reply:
I'm not sure that we need both `setFilter` and `addFilters` since their functionality is so similar. If we were to have an `addFilters`, I would expect a `removeFilters` as well.  I'm wondering if it would make more sense to have just a single `setFilters` which would allow for setting everything at once.

Reply:
That could work but I believe this would mean that all static filters would need get the current filter.static state and merge it with its new set of filters (which would cause it to subscribe to the entire static filterstate). So say we just have something like `setFilters(Record<string, SelectableFilter[]> 
 null)` which will directly set to be state.filter.static. Then a StaticFilter component would do something like:
 ```
 const staticFilters = useAnswersState(state => state.filter.static);
 useAnswersAction().setFilters({...staticFilters, someFilterId: someListOfFilters});
 ```

Reply:
Update: keeping `setFilter` (now changed to `setStaticFilters`) as it's better to use this when applying filters through url instead of calling multiple toggle functions for each filter. Removed `addFilters`